[SparkR][Minor] Add more examples to window function docs

junyangq opened the pull request:
## What changes were proposed in this pull request?

This PR adds more examples to window function docs to make them more accessible to the users.

It also fixes default value issues for `lag` and `lead`.
## How was this patch tested?

Manual test, R unit test.

junyangq committed:
Add more examples to window function docs.

SparkQA commented:
**[Test build #64327 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64327/consoleFull)** for PR 14779 at commit [`fe76c69`](https://github.com/apache/spark/commit/fe76c69f78721e8825dbf4b27af728a147102c72).

SparkQA commented:
**[Test build #64327 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64327/consoleFull)** for PR 14779 at commit [`fe76c69`](https://github.com/apache/spark/commit/fe76c69f78721e8825dbf4b27af728a147102c72).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/64327/
Test PASSed.

felixcheung commented on R/pkg/R/functions.R:
```diff
@@ -3239,7 +3259,15 @@ setMethod("lead",
+#'   # Partition by Species and order by Sepal_Length
+#'   ws <- orderBy(windowPartitionBy("Species"), "Sepal_Length")
+#'
+#'   # Get ntil group id (1-4) for Sepal_Length
```
ntile?

felixcheung commented:
btw, you might want to avoid using `iris` for example. There has been several bugs where people would like to address the limitation with column names with `.` - if that is fixed then the example would be wrong to reference column as `Sepal_Length` instead of its original name `Sepal.Length`

felixcheung commented on R/pkg/R/functions.R:
```diff
@@ -3200,19 +3212,27 @@ setMethod("lag",
 #'
 #' @param x Column to compute on
-#' @param offset Number of rows to offset
+#' @param offset Number of rows to offset. If not specified, the default is 1.
```
perhaps explain more what offset/defaultValue will do like in Python?

junyangq replied:
This is actually in the description?

felixcheung replied:
in @rdname lag, there's a bit more information on these @param. Perhaps here with @rdname lead we could do similar?

junyangq committed:
Change example dataset of window function to mtcars.

This is to avoid potential issue that will be caused if we fix the interpretation of . to _ in column names.

SparkQA commented:
**[Test build #64363 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64363/consoleFull)** for PR 14779 at commit [`32cf76c`](https://github.com/apache/spark/commit/32cf76cbe5fbd2bf8201f6377ad89eae7e93824e).

junyangq committed:
Fix inaccurate description of lag function.

SparkQA commented:
**[Test build #64365 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64365/consoleFull)** for PR 14779 at commit [`7ac95d5`](https://github.com/apache/spark/commit/7ac95d5a65936d6a4a9d58db1babd4dd021ed996).

felixcheung commented on R/pkg/R/functions.R:
```diff
@@ -3194,25 +3206,34 @@ setMethod("lag",
+#'   # Partition by am (transmission) and order by hp (horsepower)
+#'   ws <- orderBy(windowPartitionBy("am"), "hp")
+#'
+#'   # Lag mpg values by 1 row on the partition-and-ordered table
```
lag -> lead?

junyangq committed:
Add more explanation of lead parameters.

felixcheung commented:
LGTM. waiting for Jenkins.

SparkQA commented:
**[Test build #64366 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64366/consoleFull)** for PR 14779 at commit [`7324a89`](https://github.com/apache/spark/commit/7324a89a664092b886bd60eae49254d71dc28b45).

SparkQA commented:
**[Test build #64363 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64363/consoleFull)** for PR 14779 at commit [`32cf76c`](https://github.com/apache/spark/commit/32cf76cbe5fbd2bf8201f6377ad89eae7e93824e).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/64363/
Test PASSed.

SparkQA commented:
**[Test build #64365 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64365/consoleFull)** for PR 14779 at commit [`7ac95d5`](https://github.com/apache/spark/commit/7ac95d5a65936d6a4a9d58db1babd4dd021ed996).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/64365/
Test PASSed.

SparkQA commented:
**[Test build #64366 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/64366/consoleFull)** for PR 14779 at commit [`7324a89`](https://github.com/apache/spark/commit/7324a89a664092b886bd60eae49254d71dc28b45).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/64366/
Test PASSed.
